### PR TITLE
Collecting etcd metrics

### LINKED ISSUE #64030: Start collecting etcd metrics in our performance tests
This is a discussion that came up few times already in the past. It'll be really helpful to have those in light of recent regressions we'd seen due to etcd changes, like:

- https://github.com/kubernetes/kubernetes/issues/60589#issuecomment-371171837
- https://github.com/kubernetes/kubernetes/issues/62064#issuecomment-384355538
- Also to help with https://github.com/kubernetes/kubernetes/issues/63245

Etcd exposes a [bunch of metrics](https://coreos.com/etcd/docs/latest/metrics.html), out of which few that're most interesting to us IMO are:

- backend_commit_duration_seconds
- snapshot_save_total_duration_seconds
- peer_round_trip_time_seconds (probably not so important for our scalability tests that're all single-node etcd setups, except [gke-regional-performance](https://k8s-testgrid.appspot.com/sig-scalability-gke#gke-large-performance-regional) one)
- sth else?

We should:

- [x] collect those metrics through our test framework (similar to how we're doing for apiserver)
- [x] plot those metrics on our performance dashboard

cc @kubernetes/sig-scalability-misc @kubernetes/sig-api-machinery-misc @wojtek-t @jpbetz @jberkus

### PR DESCRIPTION
**Pull request by krzysied**
**What this PR does / why we need it**:
Adding etcd metrics to performance test log.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
ref #64030

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by shyamjvs** [comment]
@krzysied Let's not mark the issue as fixed in the PR description - as there are more parts to it (e.g perfdash).

File: test/e2e/framework/metrics_util.go [hunk 1]
```diff
@@ -228,6 +229,77 @@ func (l *SchedulingMetrics) PrintJSON() string {
...
+	}
+	sort.Float64s(keys)
+	lessValuesSum := 0
+	for _, k := range keys {
+		h.Buckets[k] -= lessValuesSum
+		lessValuesSum += h.Buckets[k]
+	}
+}
+
+func (h *histogram) CalculatePercs() (float64, float64, float64) {
```
Comment (shyamjvs) [inline test/e2e/framework/metrics_util.go]: Can you instead make this a function that takes single argument (like 0.5, 0.9, etc) and returns a single value for the percentile value?

Comment (krzysied) [reply test/e2e/framework/metrics_util.go]: done

**Comment by shyamjvs** [comment]
/lgtm
/hold
@wojtek-t @jpbetz - In case you may have some comments (note that this is an initial attempt to start collecting few basic but useful metrics).
We have the metrics available as histograms, but the reason why I asked for conversion to percentiles is so we can better visualize on our perf dashboard.

**Comment by shyamjvs** [comment]
/hold cancel

As no comments were raised.

**Pull request merged**